### PR TITLE
Fix tracing undefined alignment (#4201)

### PR DESCRIPTION
**Pull request by jlwehle**
Test case and fix for issue-4201.

### CODE REVIEW

File: test_regress/t/t_trace_ub_misaligned_address.pl [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        verilator_make_cmake => 0,
+        verilator_make_gmake => 0,
+        make_main => 0,
+        );
+
+    execute(
+        check_finished => 1,
+        );
+
+    files_identical("$Self->{obj_dir}/vlt_sim.log", $Self->{golden_filename}, "logfile");
```
Comment (wsnyder) [inline test_regress/t/t_trace_ub_misaligned_address.pl:32]: Is the .out file adding some sort of coverage? If not perhaps remove this and the .out?

Comment (jlwehle) [reply test_regress/t/t_trace_ub_misaligned_address.pl:32]: The intent is for the test to fail if any additional messages (such as the runtime warning about undefined behavior) appear.  Is there a way of doing that without using a .out file?

Comment (wsnyder) [reply test_regress/t/t_trace_ub_misaligned_address.pl:32]: Thats OK to have then.

File: include/verilated_trace.h [hunk 2]
```diff
@@ -447,7 +447,9 @@ class VerilatedTraceBuffer VL_NOT_FINAL : public T_Buffer {
         if (VL_UNLIKELY(diff)) fullIData(oldp, newval, bits);
     }
     VL_ATTR_ALWINLINE void chgQData(uint32_t* oldp, QData newval, int bits) {
-        const uint64_t diff = *reinterpret_cast<QData*>(oldp) ^ newval;
+        QData old;
+        memcpy(&old, oldp, sizeof(old));
```
[the remark this answers is not in the document]
Comment (jlwehle) [reply include/verilated_trace.h]: Done.

**Pull request merged**